ecosystem/web-wallet/src/pages/Login.tsx

```diff
@@ -1,30 +1,56 @@
 import React, { useState } from 'react'
-import { AptosAccount } from 'aptos'
 import { Buffer } from 'buffer'
+import { KEY_LENGTH } from '../constants'
 import { useNavigate } from 'react-router-dom'
 import { useGlobalState } from '../GlobalState'
+import { createNewAccount, loginAccount } from '../utils/account'
 
 import './App.css'
 
 export default function Login () {
   const [key, setKey] = useState('')
+  const [error, setError] = useState('')
   const [, dispatch] = useGlobalState()
   const navigate = useNavigate()
 
   function handleSubmit (event: React.FormEvent<HTMLFormElement>) {
     event.preventDefault()
-    const encodedKey = Uint8Array.from(Buffer.from(key, 'hex'))
-    const account = new AptosAccount(encodedKey, undefined)
-    dispatch({ account })
-    navigate('/wallet')
+    const result = (loginAccount(key))
+    if (result.isOk()) {
```

Comment:
Haha this is very rusty

Reply:
lol yea I felt weird doing it another way with try catches.